File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>

int coleman_lieu(string input){ // applying the coleman_lieu to output the index
    float L, S;
    int count_word=0, count_letter=0, count_sent=0;
    char word;
    string letter;
    int index;

    //running thru the letters of the input 
    for (int i = 0; input[i] != '\0'; i++){
        
        // tolower the letters
        if (isupper(input[i])){
            input[i] = tolower(input[i]);
        }

        // words counting
        if (input[i] == ' '){
            count_word ++;
        }

        //letter counting
        if (input[i] >= 'a' && input[i] <= 'z'){
            count_letter ++;
        }

        //sentence counting
        if (input[i] == '!' || input[i] == '.' || input[i] == '?'){
            count_sent++;
        }
    }

    count_word = count_word + 1; //counting spaces would work except for the last word, so by adding 1 more, we would have the right number

    L = ((float)count_letter / (float)count_word) * 100;
    S = ((float)count_sent / (float)count_word) * 100;
    index = 0.0588*L - 0.296*S -15.8 + 0.5; // to add an extra 0.5 to round up the index num;
    return index;
};

int main(){
    string text = get_string("Text: ");

    int grade = coleman_lieu(text);

    if (grade >= 16){
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else{
        printf("Grade %d\n", grade);
    }
    
}

File: pset6/readability/readability.py
from cs50 import get_string
import string

def coleman(input):
    count_letter = count_sent = 0
    count_word = 1

    for letter in input: # loop thru input with letter as a char
        
        if letter == ' ':
            count_word += 1
        
        if letter in list(string.ascii_letters):
            count_letter += 1
        
        punc = ['!','.','?']
        if letter in punc:
            count_sent += 1

    L = (count_letter/count_word) * 100
    S = (count_sent/count_word) * 100

    index = 0.0588*L - 0.296*S - 15.8 + 0.5

    return index

def main():
    text = get_string("Text: ")
    grade = int(coleman(text))

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print("Grade " + str(grade))


main()